File: app/src/main/java/com/mdmobile/cyclops/dataModels/api/Token.java
package com.mdmobile.cyclops.dataModels.api;


public class Token {

    private String access_token, token_type, refresh_token;
    private int expires_in;


    public Token(String access_token, String token_type, int expires_in, String refresh_token) {
        this.access_token = access_token;
        this.token_type = token_type;
        this.expires_in = expires_in;
        this.refresh_token = refresh_token;
    }

    public String getAccess_token() {
        return access_token;
    }

    public String getToken_type() {
        return token_type;
    }

    public int getTokenExpiration() {
        return expires_in;
    }

    public String getRefreshToken() {
        return refresh_token;
    }
}

File: app/src/main/java/com/mdmobile/cyclops/dataModel/api/newDataClass/Token.kt
package com.mdmobile.cyclops.dataModel.api.newDataClass

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import com.mdmobile.cyclops.provider.McContract

@Entity(tableName = McContract.TOKEN_TABLE_NAME)
data class Token(
        @PrimaryKey
        @field:SerializedName("access_token")
        var token: String = "N/A",
        @Ignore
        val token_type: String = "N/A",
        @Ignore
        val tokenExpiration: Int= -1,
        @Ignore
        val refreshToken: String= "N/A")
